### PR TITLE
Adding function to return list (vector) of MT numbers in a syntaxTree::File.

### PR DESCRIPTION
**Pull request by jlconlin**
This makes it possible to get all the MT numbers in the File without the client having to iterate through everything.

### CODE REVIEW

File: src/ENDFtk/syntaxTree/File.hpp [hunk 1]
```diff
@@ -52,4 +52,7 @@ class File {
   int MF() const { return this->fileNo; }
 
   int fileNumber() const { return this->MF(); }
+
+  std::vector< int > sectionNumbers() const {
```
Comment (apmccartney) [inline src/ENDFtk/syntaxTree/File.hpp:56]: Now why would you take a nice lazy view and force the client to heap allocate memory for a container?

```suggestion
  auto sectionNumbers() const { 
```

File: src/ENDFtk/syntaxTree/File/test/File.test.cpp [hunk 2]
```diff
@@ -29,10 +29,16 @@ SCENARIO( "Creating a syntax tree of an ENDF File" ){
           < syntaxTree::File< std::string::iterator > >
           ( head, start, beg, end, lin );
         
+        THEN( "We can exstract the MT numbers" ){
+          std::vector< int > refMTs{ 1, 2, 102 };
+          CHECK( refMTs == original->sectionNumbers() );
```
Comment (apmccartney) [inline src/ENDFtk/syntaxTree/File/test/File.test.cpp:34]: related to previous comment

```suggestion
          CHECK( ranges::equal( refMTs,original->sectionNumbers() ) );
```

**Comment by jlconlin** [comment]
Thanks @apmccartney for the comments. (Glad to see you are still alive!) I have addressed them in my latest changes.

**Review by whaeck — APPROVED** [review]
I agree with @apmccartney comments so now we're good to go.

**Pull request merged**